### PR TITLE
Few Stock Suggestions and column name changes.

### PR DESCRIPTION
**Pull request by bsmaheshkumar**
Implemented review from @coderhindu 

screen shot 2018-01-07 at 1 57 34 am

Support for My Investment View

screen shot 2018-01-06 at 4 42 19 pm
screen shot 2018-01-06 at 4 42 12 pm

 
screen shot 2018-01-06 at 3 03 57 am

1. Added Suggestions : TANLA and SUZLON
2. Fixed sorting issue by Days Column
3. Updated the Column names and alignment
4. Support for Company name.

TBD: Company Name to be displayed has Tooltip, when user moves on the
scripts. (This is not working.)

### CODE REVIEW

File: wealth-management-client/src/app/app.component.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { Component } from '@angular/core';
+import { TabViewModule } from 'primeng/primeng';
```
Comment (coderhindu) [inline wealth-management-client/src/app/app.component.ts]: This is not required.

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { Component } from '@angular/core';
```
Comment (coderhindu) [inline wealth-management-client/src/app/reports/userviews/userview-grid.component.ts]: portfolio/portfolio.component.ts

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<p-dataTable [immutable]="false" [value]="userInvestmentList" [sortMode]="multiple" [responsive]="true">
```
Comment (coderhindu) [inline wealth-management-client/src/app/reports/userviews/userview-grid.component.html]: portfolio.component.html

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.progress {
```
Comment (coderhindu) [inline wealth-management-client/src/app/reports/userviews/userview-grid.component.css]: Move this style to styles.css and delete both css files of components.

File: wealth-management-client/src/app/reports/reports.module.ts [hunk 5]
```diff
@@ -2,21 +2,26 @@ import { NgModule } from '@angular/core';
...
   imports: [
     DataTableModule,
+    TooltipModule,
     CommonModule,
     HttpModule
   ],
   exports: [
-    RecommendationGridComponent
+    RecommendationGridComponent,
+    UserviewGridComponent
```
Comment (coderhindu) [inline wealth-management-client/src/app/reports/reports.module.ts]: PortfolioComponent

File: wealth-management-client/src/app/reports/recommendations/recommendation-grid.component.ts [hunk 6]
```diff
@@ -24,10 +24,12 @@ export class RecommendationGridComponent {
       this.mpservice.getCMPForScripts(data.script).subscribe(value => {
         let stock: Stock = new Stock();
         stock.marketPrice = value.quoteSummary.result[0].price.regularMarketPrice.raw;
```
Comment (coderhindu) [inline wealth-management-client/src/app/reports/recommendations/recommendation-grid.component.ts]: Remove all debug logs

File: wealth-management-client/src/app/modal/mystock.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+export class MyStock {
```
Comment (coderhindu) [inline wealth-management-client/src/app/modal/mystock.ts]: Don't need another modal object.

Comment (coderhindu) [reply wealth-management-client/src/app/modal/mystock.ts]: `export class Stock {

  /* form data */
  script: string;
  name: string;
  investPrice: number;
  rcomPrice: number;
  targetPrice: number;
  marketPrice: number;
  rcomDate: number;
  rcomTimeScale: string;
  rcomBy: string;

  /* calculated values */
  isUserPortfolio: boolean = false;
  targetPercentage: number;
  achivedPercentage: number;
  achivedPercentageClass: string;
  achivedPercentageWidth: number;
  days: string;

  constructor() {

  }

  setMarketPrice(marketPrice: number) : void {
    this.marketPrice = marketPrice;
  }

  setGridData(data: any): void {
    var firstDate = new Date(data.rcomDate);
    var secondDate = new Date();
    this.script = data.script;
    this.name = data.name;
    this.investPrice = data.buyPrice;
    this.rcomPrice = data.rcomPrice;
    this.targetPrice = data.targetPrice;
    this.targetPercentage = (((this.targetPrice - this.rcomPrice) / this.rcomPrice) * 100);
    if (this.marketPrice !== undefined && this.marketPrice != 0) {
      if (this.isUserPortfolio) {
        this.achivedPercentage = (((this.marketPrice - this.investPrice) / this.investPrice) * 100);
        this.achivedPercentageWidth = Math.abs(this.achivedPercentage);
      } else {
        this.achivedPercentage = (((this.marketPrice - this.rcomPrice) / this.rcomPrice) * 100);
        this.achivedPercentageWidth = Math.abs(this.achivedPercentage);
      }
      this.achivedPercentageClass = (this.achivedPercentage) > 0 ? 'success' : 'danger';
    }
    this.rcomDate = new Date(data.rcomDate).getTime();
    this.days = Math.abs((firstDate.valueOf() - secondDate.valueOf()) / (24 * 60 * 60 * 1000)).toFixed(0);
    this.rcomTimeScale = data.rcomTimeScale;
    this.rcomBy = data.rcomBy;
  }
}
`

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.html (hunk 3 above)
Comment (coderhindu) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.html]: achivedPercentage will be called Profit/Loss and shall be calculated with investment price.
targetPercentage will be calculated with investment price.

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  scripts: string[] = new Array<string>();
+
+  constructor() {
+    this.getMyInvestmentList();
+  }
+
+  getMyInvestmentList() {
+    let myStocksData = JSON.parse(JSON.stringify(myStocks));
+    myStocksData.map(data => {
+        let mystock: MyStock = new MyStock();
```
Comment (coderhindu) [inline wealth-management-client/src/app/reports/userviews/userview-grid.component.ts]: let stock: Stock = new Stock();
stock.isUserPortfolio = true;
stock.setMarketPrice(value.quoteSummary.result[0].price.regularMarketPrice.raw);
stock.setGridData(data);

Comment (coderhindu) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.ts]: Get recommended data also while iterating over myStockList.

File: wealth-management-client/src/app/data/stocks-recommendations.json [hunk 9]
```diff
@@ -39,15 +39,6 @@
     "rcomBy": "Hemanth",
```
Comment (coderhindu) [inline wealth-management-client/src/app/data/stocks-recommendations.json]: Please change AVANTI to AVANTIFEED.
SHARDA to SHARDAMOTR
PREMCO is not listed in NSE

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.ts (hunk 2 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.ts]: ok, will change.

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.html (hunk 3 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.html]: achivedPercentage based on the recommended view.
Looks like the understanding of the US is different between us.
Lets have a call and sync-up.

Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.html]: Profit/Loss is specific to user view.

targetPercentage/achivedPercentage is based on the recommendation.

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.css (hunk 4 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.css]: OK.
So will we have only one styles.css for the entire app?

File: wealth-management-client/src/app/reports/reports.module.ts (hunk 5 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/reports.module.ts]: Shall rename to ProtfolioGridComponent

File: wealth-management-client/src/app/modal/mystock.ts (hunk 7 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/modal/mystock.ts]: We need other model.
We cannot mix targetPercentage and achivedPercentage based on the user investment.
The idea is to have 2 different view.
1. Recommendation View
2. Portfolio View

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.ts (hunk 8 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.ts]: Looks like the User story understanding is different between both of us. Lets have a call and sync-up
This check-in does not include all the functionality.
Did you check the TBD list in my comments?

File: wealth-management-client/src/app/data/stocks-recommendations.json (hunk 9 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/data/stocks-recommendations.json]: OK.
Do we need a different query for BSE stock only? If yes, lets add it to backlog.

File: wealth-management-client/src/app/app.component.ts (hunk 1 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/app.component.ts]: Done.

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.ts (hunk 2 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.ts]: Done.

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.html (hunk 3 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.html]: Changes based on our sync-up

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.css (hunk 4 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.css]: Done

File: wealth-management-client/src/app/reports/reports.module.ts (hunk 5 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/reports.module.ts]: Done.

File: wealth-management-client/src/app/reports/recommendations/recommendation-grid.component.ts (hunk 6 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/recommendations/recommendation-grid.component.ts]: Done.

File: wealth-management-client/src/app/reports/userviews/userview-grid.component.ts (hunk 8 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/reports/userviews/userview-grid.component.ts]: Done.

File: wealth-management-client/src/app/data/stocks-recommendations.json (hunk 9 above)
Comment (bsmaheshkumar) [reply wealth-management-client/src/app/data/stocks-recommendations.json]: Done.
Have removed PREMCO from now.

**Pull request merged**